doc/manual/src/architecture/architecture.md

```diff
@@ -0,0 +1,42 @@
+# Architecture
+
+This chapter describes how Nix works.
```

Comment:
I still slightly prefer "what Nix is", as "how Nix works" sounds to me like emphasizing implementations interfaces.

But this is just a slight preference. If you feel more strongly, keep it as is. "how _ works" is a well-known phrase, and "is" is a weak, unevocative verb.

Reply:
My reasons may be mundane, but after reading our overview (especially without the addition from https://github.com/nix-community/nix/pull/9) I still wouldn't be able to tell what Nix *is*. But maybe how it works.

Saying "X is Y" implies an identity, and what would we identify Nix with other than its own definition, possibly in terms of its data types and operations?

We could very much have a subsection for each term that provides exactly that. Then you'd have a prose description, a graphical illustration, and a formal textual specification. You already gave one for file system objects. But this is longer-term thinking, right now I suggest we focus on prose, because that covers the broadest audience.